src/v/test_utils/async.h

```diff
@@ -50,4 +51,33 @@ requires ss::ApplyReturns<Predicate, bool> ||
           });
       }));
 }
+
+// When a test expects that any background fibers should complete promptly,
+// and wants to send a barrier through all the inter-CPU queues to ensure
+// that earlier-submitted tasks have reached their destination cores already.
+//
+// This is useful in tests that know they have put the system into a state where
+// it will get to a known state once all the non-i/o-blocking tasks in flight
+// have completed, such as background release of quota units.
+//
+// **Be aware** that there are assumptions to using this:
+// A) That your test code is using the same default scheduling group that
+//    this routine will run within.
+// B) That debug mode scheduling randomization in seastar does not re-order
+//    things so dramatically that our messages can go between every core and
+//    back before a future that was already ready on some shard gets run.
+void flush_tasks() {
+    // Ensure anything in inter-CPU queues before we entered the function
+    // has drained: this is an all-to-all to cover the full mesh of queues
+    // between cores.
+    ss::smp::invoke_on_all([]() {
+        return ss::smp::invoke_on_all(
+          []() { return ss::make_ready_future<>(); });
```

Comment:
I this works then great. one thought I had was that if randomization in debug mode happens when selecting a new ready-to-run future you could return `ss::yield` here (which I think is the replacement for `ss::later()`) which isn't ready to run but is, according to the docs, "ready to run very soon", which ostensibly would allow at least one round of futures to be selected before this one. but it all seems probabilisitic anyway so it likely doesn't matter much either way...

Reply:
Good idea: substituted yield for make_ready_future